cmd/routedns/main.go

```diff
@@ -244,15 +244,15 @@ func start(opt options, args []string) error {
 			if err != nil {
 				return err
 			}
-			httpProxyAddr := net.ParseIP(l.DoH.HTTPProxyAddr)
-			if l.DoH.HTTPProxyAddr != "" && httpProxyAddr == nil {
-				return fmt.Errorf("listener '%s' proxy-address '%s' is not an IPv4 or IPv6 address", id, l.DoH.HTTPProxyAddr)
+			_, httpProxyNet, err := net.ParseCIDR(l.Frontend.HTTPProxyNet)
+			if l.Frontend.HTTPProxyNet != "" && httpProxyNet == nil {
+				return fmt.Errorf("listener '%s' trusted-proxy '%s': %v", id, l.Frontend.HTTPProxyNet, err)
```

Comment:
While this works as expected, it's a little hard to read due the unusual error handling. How about something like this, a little more explicit:

```go
	var httpProxyNet *net.IPNet
	if l.Frontend.HTTPProxyNet != "" {
		_, httpProxyNet, err = net.ParseCIDR(l.Frontend.HTTPProxyNet)
		if err != nil {
			return fmt.Errorf("listener '%s' trusted-proxy '%s': %v", id, l.Frontend.HTTPProxyNet, err)
		}
	}
```